### ISSUE TITLE
[BUG] GiveNamedItem parameters are incorrect (?)

### ISSUE THREAD

**Issue by K4ryuu** [roflmuffin/CounterStrikeSharp#466]
`GiveNamedItem` function returns wrong values to the `VirtualFunctions.GiveNamedItemFunc.Hook`

**In the older versions it also returned the classname properly.**

When you purchase and item or give by console cmd with sv_cheats and you log the first three params of the DynamicHook you see this (which is correct):
```
0. �9�P
1. weapon_m4a1
2. �νP
```

But when you give an item with GiveNamedItem CSS functions, you see this (and you cannot get the classname of weapon)
```
0. �9�P
1. @6Y�\U
2. Empty (null)
```

In my case I want to use the classname for further validations, but its just like not exists if GiveNamedItem is being used to give weapon

The code I added at the first lines of GiveNamedItemFunc.Hook:

```cs
for (int i = 0; i < 3; i++)
{
      var paramValue = hook.GetParam<string>(i);
      if (string.IsNullOrWhiteSpace(paramValue))
      {
	      Logger.LogWarning($"{i}. Empty");
      }
      else
      {
	      Logger.LogWarning($"{i}. {paramValue}");
      }
}
```

**Comment by ianlucas** [comment]
By "older versions" you mean pre-Fire Sale update, right? I also use that hook in one my plugins and updated the signature for CSSharp via #463. Previously the hook was not being triggered as I think we were hooking another function that also gives items (but that is not called when purchasing/receiving items on spawn).

**Comment by K4ryuu** [comment]
It was working completely fine until the latest big update that broke the signs. Since then only the item purchase and console give triggers it well. This could be possibly caused by dynohook in CSS(?), but I'm not sure

**Comment by ianlucas** [comment]
We're using offsets for calling GiveNamedItem, but I think memory function hooking currently just works for signatures, that's why CSSharp still maintains the GiveNamedItem signature.

**Comment by ianlucas** [comment]
> It was working completely fine until the latest big update that broke the signs. Since then only the item purchase and console give triggers it well. This could be possibly caused by dynohook in CSS(?), but I'm not sure

Yeah I'm also not sure if this is related to DynoHook, but I'd guess it's not.

**Comment by roflmuffin** [comment]
@K4ryuu Can you confirm if there are still issues with this?

**Comment by K4ryuu** [comment]
![CleanShot 2024-07-27 at 20 05 46@2x](https://github.com/user-attachments/assets/73af29c2-7f86-4edb-8e2f-f0e5c89a69ca)

Weapons are visible again. Perfect job! Issue is solved.
Thanks for your work!

**Comment by K4ryuu** [comment]
Apologise, but I was wrong with the feedback as I checked on player spawn for some reason. On spawn it works well, but when you use GiveNamedItem, it still returns the following data:

![CleanShot 2024-08-07 at 00 55 52@2x](https://github.com/user-attachments/assets/5bb4953d-3d65-4cb8-8029-b4f943882306)

**Comment by K4ryuu** [comment]
Seems like it's working perfectly since then. I don't know when it got fixed exactly
